cockatrice/src/client/ui/widgets/visual_deck_editor/visual_deck_editor_widget.cpp

```diff
@@ -0,0 +1,347 @@
+        zoneContainer->findChildren<DeckCardZoneDisplayWidget *>();
+    for (const QString &zone : *deckListModel->getZones()) {
+        bool found = false;
+        for (DeckCardZoneDisplayWidget *displayWidget : cardZoneDisplayWidgets) {
+            if (displayWidget->zoneName == zone) {
+                found = true;
+            }
+        }
+
+        if (!found) {
+            DeckCardZoneDisplayWidget *zoneDisplayWidget = new DeckCardZoneDisplayWidget(
+                zoneContainer, deckListModel, zone, activeGroupCriteria, activeSortCriteria, 20, 10, cardSizeWidget);
+            connect(zoneDisplayWidget, &DeckCardZoneDisplayWidget::cardHovered, this, &VisualDeckEditorWidget::onHover);
+            connect(zoneDisplayWidget, &DeckCardZoneDisplayWidget::cardClicked, this,
+                    &VisualDeckEditorWidget::onCardClick);
+            connect(this, &VisualDeckEditorWidget::activeSortCriteriaChanged, zoneDisplayWidget,
+                    &DeckCardZoneDisplayWidget::onActiveSortCriteriaChanged);
+            connect(this, &VisualDeckEditorWidget::activeGroupCriteriaChanged, zoneDisplayWidget,
+                    &DeckCardZoneDisplayWidget::onActiveGroupCriteriaChanged);
+            connect(this, &VisualDeckEditorWidget::displayTypeChanged, zoneDisplayWidget,
+                    &DeckCardZoneDisplayWidget::refreshDisplayType);
+            zoneContainerLayout->addWidget(zoneDisplayWidget);
+        }
+    }
+}
+
+void VisualDeckEditorWidget::deleteZoneIfDoesNotExist()
+{
+    QList<DeckCardZoneDisplayWidget *> cardZoneDisplayWidgets =
+        zoneContainer->findChildren<DeckCardZoneDisplayWidget *>();
+    for (DeckCardZoneDisplayWidget *displayWidget : cardZoneDisplayWidgets) {
+        bool found = false;
+        for (const QString &zone : *deckListModel->getZones()) {
+
+            if (displayWidget->zoneName == zone) {
+                found = true;
+            }
+        }
+
+        if (!found) {
+            zoneContainerLayout->removeWidget(displayWidget);
+        }
+    }
+}
+
+void VisualDeckEditorWidget::resizeEvent(QResizeEvent *event)
+{
+    QWidget::resizeEvent(event);
+    zoneContainer->setMaximumWidth(scrollArea->viewport()->width());
+}
+
+void VisualDeckEditorWidget::actChangeActiveGroupCriteria()
+{
+    activeGroupCriteria = groupByComboBox->currentText();
+    emit activeGroupCriteriaChanged(activeGroupCriteria);
+}
+
+void VisualDeckEditorWidget::actChangeActiveSortCriteria()
+{
+    QStringList selectedCriteria;
+    for (int i = 0; i < sortByListWidget->count(); ++i) {
+        QListWidgetItem *item = sortByListWidget->item(i);
+        selectedCriteria.append(item->text()); // Collect user-defined sort order
+    }
+
+    activeSortCriteria = selectedCriteria;
+
+    emit activeSortCriteriaChanged(selectedCriteria);
+}
+
+void VisualDeckEditorWidget::decklistDataChanged(QModelIndex topLeft, QModelIndex bottomRight)
+{
+    // Might use these at some point.
+    (void)topLeft;
```

Comment:
We should be more consistent with Q_UNUSED(), (void), and /* comment out input param */. Note for the future.